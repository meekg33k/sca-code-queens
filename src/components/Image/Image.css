.imageClass {
  border-radius: 100px;
}