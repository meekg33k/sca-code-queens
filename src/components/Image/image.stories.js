import React from "react";
import {storiesOf} from "@storybook/react";
import {withInfo} from "@storybook/addon-info";
import {withKnobs, text} from "@storybook/addon-knobs";
import Image from "./index";

storiesOf("Image", module)
    .addDecorator(
        withInfo({
            text: "This is an Image component"
        })
    )

    .addDecorator(
        withKnobs()
    )

    .add("An image", () => (
        <Image
            source = {text("path", "./assets/smiley.jpg")}
            alt={text("alt", "DJ Spinal")}
            
        />
    ))

    .add("Image", () => (
        <Image
            source = {text("path", "./link/to/image")}
            alt={text("alt", "this is an image")}
            
        />
    ));

    