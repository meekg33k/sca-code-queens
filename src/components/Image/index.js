import React from "react";
import PropTypes from "prop-types";
import "./Image.css";

const Image = props => {
  const { source, height, width, alt } = props;
  return (
    //<div className={`${source} ${height} ${width} ${alt}`}>
    <div className="imageDiv">
      <img className="imageClass" src={source} height={height} width={width} alt={alt} />
    </div>
  );
};

Image.defaultProps = {
  alt: "image",
  source: "someimage",
  height: 300,
  width: 300
};

Image.propTypes = {
  alt: PropTypes.string,
  source: PropTypes.string,
  height: PropTypes.number,
  width: PropTypes.number
};

export default Image;
