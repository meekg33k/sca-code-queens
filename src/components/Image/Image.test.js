import React from "react";
import { shallow } from "enzyme";
import Image from "./index";

describe("Image Component render", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<Image />)
        expect(wrapper).toMatchSnapshot();
    });

    it("renders an image", () => {
        const wrapper = shallow(<Image />)
            const findImage = wrapper.find("img")
            expect (findImage.length).toBe(1)
    })

})