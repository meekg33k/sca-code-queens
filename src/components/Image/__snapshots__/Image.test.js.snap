// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component render renders correctly 1`] = `ShallowWrapper {}`;
